arviz/data/inference_data.py

```diff
@@ -80,15 +80,20 @@ def to_netcdf(self, filename, compress=True):
             Location of netcdf file
         """
         mode = "w"  # overwrite first, then append
-        for group in self._groups:
-            data = getattr(self, group)
-            kwargs = {}
-            if compress:
-                kwargs["encoding"] = {var_name: {"zlib": True} for var_name in data.variables}
-            data.to_netcdf(filename, mode=mode, group=group, **kwargs)
-            data.close()
-            mode = "a"
-        return filename
+        if self._groups:  # check's whether a group is present or not.
+            for group in self._groups:
+                data = getattr(self, group)
+                kwargs = {}
+                if compress:
+                    kwargs["encoding"] = {var_name: {"zlib": True} for var_name in data.variables}
+                data.to_netcdf(filename, mode=mode, group=group, **kwargs)
+                data.close()
+                mode = "a"
+                return filename
```

Comment:
here is indentation error.

You could return `filename` at the end of the function.